csharp/ql/src/Stubs/make_stubs.py

```diff
@@ -0,0 +1,114 @@
+# Tool to generate 'stub.cs' files inside C# qltest projects.
+# The purpose of this is to stub out assemblies from the qltest case
+# so that the test is self-contained and will still run without them.
+#
+# To do this, it
+# 1. Performs a regular qltest to generate a snapshot
+# 2. Runs a QL query on the snapshot to find out which symbols are needed,
+#    then uses QL to generate a string of C# code.
+# 3. Re-runs the test to ensure that it still compiles and passes.
+
+import sys
+import os
+
+print('Script to generate stub.cs files for C# qltest projects')
+
+if len(sys.argv) < 2:
+    print("Please supply a qltest directory.")
+    exit(1)
+
+testDir = sys.argv[1]
+
+if not os.path.isdir(testDir):
+    print("Directory", testDir, "does not exist")
+    exit(1)
+
+# Does it contain a .ql file and a .cs file?
+
+foundCS = False
+foundQL = False
+
+for file in os.listdir(testDir):
+    if file.endswith(".cs"):
+        foundCS = True
+    if file.endswith(".ql") or file.endswith(".qlref"):
+        foundQL = True
+
+if not foundQL:
+    print("Test directory does not contain .ql files. Please specify a working qltest directory.")
+    exit(1)
+
+if not foundCS:
+    print("Test directory does not contain .cs files. Please specify a working qltest directory.")
+    exit(1)
+
+cmd = 'odasa selfTest'
+print(cmd)
+if os.system(cmd):
```

Comment:
I would normally use `subprocess.check_call` instead.

Reply:
That's much better, thanks.